refactor(core): add tslint rule entry-point for static-query migration

devversion opened the pull request:
In order to be able to use the static-query migration logic within
Google, we need to provide a TSLint rule entry-point that wires up
the schematic logic and provides reporting and automatic fixes.

kara reviewed (approved):
LGTM for query-related code

devversion committed:
refactor(core): add tslint rule entry-point for static-query migration

In order to be able to use the static-query migration logic within
Google, we need to provide a TSLint rule entry-point that wires up
the schematic logic and provides reporting and automatic fixes.

devversion committed:
refactor(core): better failure message for explicit-query timing tslint rule

Improves the failure message for the `explicit-query` timing TSLint rule
that is used within Google. Currently it's not very clear what action
developers need to take in order to resolve the lint failure manually.

devversion commented:
Marked as ready for review, as I restructured the schematic slightly to make it easier to sync this into G3. e.g. not using the devkit in the tslint rule.

We still need to talk about performance if the rule is used in IDE's. If that's case, then the ideal solution would be to just disable the query usage analysis when not in TSLint fix-mode. Though I'm not entirely sure if it's possible to detect if TSLint runs in fix-mode or not..
